scanner.cs

```diff
@@ -3,66 +3,176 @@
 using System.Collections.Generic;
 
 public class Scanner {
-	private int curByte = 0;
-	private byte[] __bytes;
-	private List<Tuple<string, TOKENS>> __tokens;
+    private int curByte = 0;
+    private int column = 1;
+    private int line = 1;
+    private char[] __bytes;
+    private List<Tuple<string, TOKENS>> __tokens;
 
-	// Initializes the scanner and checks for file format
-	public void initializeScanner(string fileName) {
-		try {
-			if(!fileName.EndsWith(".mp")) {
-				// File format exception
-				throw new Exception(Constants.ERROR_FILE_FORMAT);
-			} else {
-				// Grab all bytes from the file
-				this.__bytes = File.ReadAllBytes(fileName);
+    // Initializes the scanner and checks for file format
+    public void initializeScanner(string fileName) {
+        try {
+            if(!fileName.EndsWith(".mp")) {
+                // File format exception
+                throw new Exception(Constants.ERROR_FILE_FORMAT);
+            } else {
+                // Grab all bytes from the file
+                StreamReader reader = new StreamReader(fileName);
+                string temp = reader.ReadToEnd();
+                __bytes = temp.ToCharArray();
```

Comment:
Good alternative; was going to suggest we still store the whole file into a buffer but I see you already did that here.